service-admin/internal/server/data/activation_key.go

```diff
@@ -73,7 +73,7 @@ func (aks *OnlineActivationKeyService) GetActivationKeyFromCodes(ctx context.Con
 	hasher.Write(jsonStr)
 	shaHash := hex.EncodeToString(hasher.Sum(nil))
 
-	credentials, err := aks.config.Credentials.Retrieve(context.Background())
+	credentials, err := aks.config.Credentials.Retrieve(context.TODO())
```

Comment:
TODO? Is that a thing or a typo?

Reply:
It is a thing and makes no difference but I'll change it back anyway!
```suggestion
	credentials, err := aks.config.Credentials.Retrieve(context.Background())
```